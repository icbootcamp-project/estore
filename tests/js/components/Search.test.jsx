import React from 'react';
import Search from '../../../src/js/components/Search';

//Below are related to enzyme
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

describe('Search Component', () => {
  let getVideos, wrapper;

  beforeEach(() => {
    getVideos = jest.fn();
    wrapper = shallow(<Search getVideos={getVideos} />);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('runs getVideos on input handler', () => {
    wrapper.find('input').simulate('change', { target: { value: 'java' } });
    expect(getVideos.mock.calls[0]).toEqual(['java']);
  });
});
