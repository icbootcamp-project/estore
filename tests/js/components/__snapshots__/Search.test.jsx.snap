// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component renders 1`] = `
<div
  className="search col-md-6 col-md-offset-4 form-inline"
>
  <input
    className="form-control"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="btn"
  >
    <span
      className="glyphicon glyphicon-search "
    />
  </button>
</div>
`;
